sql-plugin/src/main/scala/org/apache/spark/sql/rapids/HashFunctions.scala

```diff
@@ -34,3 +39,44 @@ case class GpuMd5(child: Expression)
     }
   }
 }
+
+case class GpuMurmur3Hash(child: Seq[Expression]) extends GpuExpression {
+  override def dataType: DataType = IntegerType
+
+  override def toString: String = s"hash($child)"
+  def nullable: Boolean = children.exists(_.nullable)
+  def children: Seq[Expression] = child
+
+  def columnarEval(batch: ColumnarBatch): Any = {
+    var nullStrScalar: Scalar = null
+    val rows = batch.numRows()
+    val childEvals: ArrayBuffer[Any] = new ArrayBuffer[Any](children.length)
+    val columns: ArrayBuffer[ColumnVector] = new ArrayBuffer[ColumnVector]()
+    try {
+      children.foreach(childEvals += _.columnarEval(batch))
+      childEvals.foreach {
+        case vector: GpuColumnVector =>
+          columns += vector.getBase
+        case col => if (col != null) {
+          withResource(GpuScalar.from(col)) {
+            scalarValue => col match {
```

Comment:
Why not just use the CUDF APIs directly?

```
columns += ColumnVector.fromScalar(scalarValue, rows)
```

Reply:
When expanding from a scalar, the function requires the `DataType` be supplied. This seems to be the easiest way to extract the type, is there an easier way to extract the type from the `GpuScalar` or the `columnarEval`.